ci/ray_ci/doc/module.py

```diff
@@ -0,0 +1,63 @@
+import importlib
+import inspect
+from types import ModuleType
+from typing import Set
+
+
+class Module:
+    def __init__(self, module: str):
+        self._module = importlib.import_module(module)
+        self._visited = set()
+        self._class_apis = set()
+        self._function_apis = set()
+
+    def walk(self) -> None:
+        self._walk(self._module)
+
+    def get_class_apis(self) -> Set:
+        self.walk()
+        return self._class_apis
+
+    def get_function_apis(self) -> Set:
+        self.walk()
+        return self._function_apis
+
+    def _walk(self, module: ModuleType) -> None:
+        """
+        Depth-first search through the module and its children to find annotated classes
+        and functions.
+        """
```

Comment:
> also mean statically interpret the logic of hook functions such as this one

I thought you only need to know which hook is associated, rather than what the logic of the hook it is. for example, python's type checking tools does not need to interpret the logic.

I don't know what this is going to be used later. will leave the decision to you and @khluu . just trying to point out other possible alternatives.